### PR TITLE
Issue #35: satelite projects should use latest checkstyle as soon at is appears in maven repo

### LINKED ISSUE #35: satelite projects should use latest checkstyle as soon at is appears in maven repo
Right now we do bump a version in a few projects https://github.com/checkstyle/checkstyle/wiki/How-to-make-a-release#update-for-released-version-reference right after release.

Let's find a way to define in pom.xml the latest snapshot and most recent version. And the latest released version when we define dependencies. To let us skip such updates at all.

CI of such projects should be able to get the latest version from https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml and run projects with such version.

We can have in all such repos, a GitHub action that runs once a day check https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml, put such values in required files, and send PRs with changes (PR is necessary as we need notice is any problems happens if we introduce breaking changes)

TASK:

- [ ] Version Bump in `ivy.xml`
- [ ] Version Bump in `pom.xml`
- [ ] Version Bump in `build.gradle`

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Partially Fixes #35

### CODE REVIEW

File: .ci/bump-gradle-version.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -e
```
Comment (romani) [inline .ci/bump-gradle-version.sh:2]: apply changes from https://github.com/checkstyle/contribution/pull/635#pullrequestreview-1079599409

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .ci/bump-gradle-version.sh (hunk 1 above)
Comment (Rahulkhinchi03) [reply .ci/bump-gradle-version.sh]: Changes Done

**Review by romani — APPROVED** [review]
fixed, tested

**Pull request merged**